src/lib/y2partitioner/actions/delete_btrfs.rb

```diff
@@ -0,0 +1,83 @@
+# encoding: utf-8
+
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage"
+require "y2partitioner/actions/delete_device"
+
+module Y2Partitioner
+  module Actions
+    # Action for deleting a Btrfs filesystem
+    #
+    # @see DeleteDevice
+    class DeleteBtrfs < DeleteDevice
+      def initialize(*args)
+        super
+        textdomain "storage"
+      end
+
+    private
+
+      # Deletes the indicated filesystem (see {DeleteDevice#device})
+      def delete
+        log.info "deleting btrfs #{device}"
+        device.blk_devices.first.delete_filesystem
+      end
+
+      # @see DeleteDevice#simple_confirm_text
+      #
+      # @note The implementation in the base class relies on the #name method
+      #   of the device, which does not exist for filesystems.
```

Comment:
Well, as you mention, only `#simple_confirm` makes sense for a filesystem, so I would remove `#recursive_confirm_text_below`. Otherwise, your `#simple_confirm` re-definition is needed even after merging "my changes". My PR fixes the problem when we are deleting a device that belongs to a multi-device Btrfs (now it informs about which multi-device filesystem is going to be destroyed), but there is nothing related to this case where the delete actions acts directly over the filesystem itself.